src/Services/Utils/SQLUpgradeService.php

```diff
@@ -657,9 +657,39 @@ function upgradeFromSqlFile($filename, $path = '')
                 if ($skipping) {
                     $this->echo("<p class='text-success'>$skip_msg $line</p>\n");
                 }
+            } elseif (preg_match('/^#IfMBOEncounterNeeded/', $line)) {
+                $emptyMBOEncounters = sqlStatementNoLog("SELECT `pid` FROM `form_misc_billing_options` WHERE `encounter` IS NULL");
+                if (sqlNumRows($emptyMBOEncounters) > 0) {
+                    while ($mBORow = sqlFetchArray($emptyMBOEncounters)) {
+                        $pids[] = $mBORow['pid'];
+                    }
+                    $encStatement = "SELECT `encounter` from `form_encounter` WHERE `pid` IN (" . implode(',', array_map('intval', $pids)) . ")";
```

Comment:
Will this break if scale up to thousands or hundreds of thousands of patients?

Reply:
could break it into only certain number of patient chunks (maybe get a count of pid first, then collect them in chunks and process them)